packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -172,6 +182,18 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
       }
     }
 
+    let resourceType = props.resourceType;
+    if (resourceType) {
+      if (!new RegExp('^([a-zA-Z0-9_\\@\\-]+)$').test(resourceType)) {
+        throw new Error('`resourceType` can only contain alphanumeric characters, "_", "@", & "-"');
+      }
+      if (resourceType.length > 60) {
+        throw new Error('`resourceType` cannot be longer that 60 characters.');
+      }
```

Comment:
This logic is already implemented here:
https://github.com/aws/aws-cdk/blob/590705586eb545c2c55cea1c0733b6d5f0876765/packages/%40aws-cdk/aws-cloudformation/lib/custom-resource.ts#L225-L241

Reply:
I thought I saw that before 🤦‍♂ . Thanks for pointing it out.